Fix for a bug when searching for 'develop'

dawelter2 opened the pull request:
Fix for a bug found when trying to search for a branch named 'develop', 
Apparently one of the results was missing a key in the dictionary, so instead of access the key, i'm using `dict.get(<key>)` to return None instead of raise KeyError.

Commit:
Fix for a bug when searching for 'develop'

prusse-martin commented on test_esss_jenkins.py:
```diff
@@ -229,6 +229,17 @@ def test_bhist(jenkins_plugin, testbot, mocker, LineMatcher):
+    jenkins_plugin._fetch_job_status("dummy")
+    mocker.patch.object(jenkins_plugin,
+                        '_get_jenkins_json_request',
+                        return_value={"_class": "hudson.model.FreeStyleBuild"})
```
Was this the returned value triggering the error?

nicoddemus replied:
Yep

nicoddemus commented on test_esss_jenkins.py:
```diff
@@ -229,6 +229,17 @@ def test_bhist(jenkins_plugin, testbot, mocker, LineMatcher):
+    mocker.patch.object(jenkins_plugin,
+                        '_get_jenkins_json_request',
+                        return_value={"_class": "hudson.model.FreeStyleBuild", "result": None})
+    jenkins_plugin._fetch_job_status("dummy")
```
Assert the return status here please

dawelter2 replied:
Done!

nicoddemus commented on test_esss_jenkins.py:
```diff
@@ -229,6 +229,17 @@ def test_bhist(jenkins_plugin, testbot, mocker, LineMatcher):
+    mocker.patch.object(jenkins_plugin,
+                        '_get_jenkins_json_request',
+                        return_value={"_class": "hudson.model.FreeStyleBuild"})
+    jenkins_plugin._fetch_job_status("dummy")
```
Assert the return status here please

Commit:
Set err-bot to 5.2.0

nicoddemus commented on dev-requirements.txt:
```diff
@@ -1,4 +1,5 @@
+errbot~=5.2
 pytest
 pytest-mock
+pluggy==0.11
```
Put comments here explaining why those pinnings (for pluggy also link to the relevant issue in pluggy's repo)

Commit:
Remove pluggy and comment why pinning errbot to 5.2

Commit:
Drop Python 3.5 from Travis and include 3.7

Commit:
Ignore all warnings from markdown module

Unfortunately newest versions are issuing further warnings; we need
to ignore those until at least we upgrade errobot (#16) to see if the
warnings go away

Commit:
Pin markdown version to < 3 and add ignored warning

Commit:
Tests and move custom exception

Use this as an inner-class so it can easily be accessed by tests from the plugin object.
errbot loads plugins dynamically, so to importing this in a test will result in a different
class object than the one loaded by errbot.

* Add test to `jenkins_plugin._fetch_job_status`
* Move custom exception

nicoddemus commented on test_esss_jenkins.py:
```diff
@@ -230,14 +231,43 @@ def test_bhist(jenkins_plugin, testbot, mocker, LineMatcher):
 
 
 def test_fetch_job_status(jenkins_plugin, mocker):
```
Please make this test parametrized. 😉

dawelter2 replied:
Done, just couldn't figure out how to parametrize the status "NOT_STARTED", so created a new test.

Commit:
Create parametrize for test

Create parametrize for `test_fetch_job_status` and move the part which
test status "NOT_STARTED" to `test_fetch_job_status_not_started`.
